src/sage/rings/polynomial/multi_polynomial_ideal.py

```diff
@@ -4965,6 +4965,15 @@ def reduce(self, f):
 
             Requires computation of a Groebner basis, which can be a
             very expensive operation.
+
+        TESTS:
+
+        Check for :issue:`38560`::
+
+            sage: I.reduce(1)
+            1
+            sage: I.reduce(pi.n())  # unfortunate side effect
```

Comment:
```suggestion
            sage: I.reduce(pi.n())
```
I think this is good behavior; it ensures everything is converted to the proper parent. Although perhaps we should have coercion instead of conversion?

Reply:
Fixed, now use coercion (and the example raises error which I think makes more sense. If the user want they can do the conversion themselves)